recipes/flatbuffers/all/conanfile.py

```diff
@@ -38,32 +38,43 @@ def config_options(self):
     def _configure_cmake(self):
         cmake = CMake(self)
         cmake.definitions["FLATBUFFERS_BUILD_TESTS"] = False
-        cmake.definitions["FLATBUFFERS_BUILD_SHAREDLIB"] = self.options.shared
-        cmake.definitions["FLATBUFFERS_BUILD_FLATLIB"] = not self.options.shared
+        cmake.definitions["FLATBUFFERS_BUILD_SHAREDLIB"] = self.options.shared and not self.options.header_only
```

Comment:
Revert it, as you don't use cmake for header-only, you don't need to customize the definitions

Reply:
Good point will do!

Reply:
Fixed